modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java

```diff
@@ -144,8 +142,8 @@ public void setProperty(Property property) {
         }
     }
 
-    public String getDefaultValue() {
-        return defaultValue;
+    public Object getDefaultValue() {
```

Comment:
The signature of this method should not be changed for backward compatibility. If you need to get object type instead of string value just use the `Object getDefault()` method.
Handling of default values has been fixed in version `1.5.4` (PR #1378, unit test for boolean values looks as [follows](https://github.com/swagger-api/swagger-core/pull/1378/files#diff-f64eab520f5cc6c19526a79d67f5d444R24)) - could you please check the version of the `swagger-core` artifact in your project?